src/cls/tabular/skyroot.fbs

```diff
@@ -0,0 +1,107 @@
+namespace Tables;
+
+union Relation { 
+  Rows,
+  Col,
+  ColInt,
+  ColFloat,
+  ColString
+}
+
+union Data { 
+  IntData,
+  FloatData,
+  StringData
+}
+
+table Root {
+  schema       : [ ubyte ] ;
+  relationData : Relation  ;
+}
+
+table Rows {
+  skyhook_version : ubyte ;  // skyhook version
+  schema_version  : ubyte ;
+
+  table_name : string     ;
+  schema     : [ string ] ;
+  nrows      : uint64     ;
+  ncols      : uint64     ;
+  layout     : string     ;
+
+  RIDs : [ uint64 ] ;
+  data : [ Record ] ;
+}
+
+table Record {
+  data: [ Data ] ;
+}
+
+table Col {
+  skyhook_version : ubyte ;  // skyhook version
+  schema_version  : ubyte ;
+
+  col_name  : string ;
+  col_index : ubyte  ;
+  nrows     : ubyte  ;
+
+  RIDs : [ uint64 ] ;
+  data : Data    ;
+}
+
+table IntData {
+  data : [ uint64 ] ;
+}
+table FloatData {
+  data : [ float ] ;
+}
+table StringData {
+  data : [ string ] ;
+}
+
+table ColInt {
+  skyhook_version : ubyte ;  // skyhook version
+  schema_version  : ubyte ;
+
+  col_name  : string ;
+  col_index : ubyte  ;
+  nrows     : ubyte  ;
+
+  RIDs : [ uint64 ] ;
+  data : [ uint64 ] ;
+}
+
+table ColFloat {
+  skyhook_version : ubyte ;  // skyhook version
+  schema_version  : ubyte ;
+
+  col_name  : string ;
+  col_index : ubyte  ;
+  nrows     : ubyte  ;
+
+  RIDs : [ uint64 ] ;
+  data : [ float ] ;
+}
+
+table ColString {
+  skyhook_version : ubyte ;  // skyhook version
+  schema_version  : ubyte ;
+
+  col_name  : string ;
+  col_index : ubyte  ;
+  nrows     : ubyte  ;
+
+  RIDs : [ uint64 ] ;
+  data : [ string ] ;
+}
+
+root_type Root ;
+
+// CODES for schema/data_type
+// 0 --> uint64
+// 1 --> float
+// 2 --> string
+
+// CODES for layout
```

Comment:
also let's create an enum for these if we do not have one that works for your case yet.